### PR TITLE
Add support for showing global vgtid executed

### PR DESCRIPTION
**Pull request by harshit-gangal**
Signed-off-by: Harshit Gangal <harshit@planetscale.com>

## Description

This PR adds a new query to show vgtid i.e. group all the current gtid_executed.

```
mysql> show global vgtid_executed;
+----------+------------------------------------------------------------------------------------------------+
| keyspace | global vgtid_executed                                                                          |
+----------+------------------------------------------------------------------------------------------------+
| commerce | shard_gtids:<keyspace:"commerce" shard:"0" gtid:"bfb145e3-98ec-11eb-b9a7-98af65a6dc4a:1-23" >  |
+----------+------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

```

## Related Issue(s)

- resolves https://github.com/vitessio/vitess/issues/7400

## Checklist
- [X] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [X]  Query Serving

### CODE REVIEW

File: go/vt/vtgate/engine/ordered_aggregate.go [hunk 1]
```diff
@@ -38,8 +37,8 @@ var _ Primitive = (*OrderedAggregate)(nil)
 // is that the underlying primitive is a scatter select with pre-sorted
 // rows.
 type OrderedAggregate struct {
-	// HasDistinct is true if one of the aggregates is distinct.
-	HasDistinct bool `json:",omitempty"`
+	// PreProcess is true if one of the aggregates is distinct.
```
Comment (systay) [inline go/vt/vtgate/engine/ordered_aggregate.go:40]: Hmmm... Why this name change?

File: go/vt/vtgate/engine/ordered_aggregate.go [hunk 2]
```diff
@@ -70,6 +69,10 @@ func (ap AggregateParams) isDistinct() bool {
 	return ap.Opcode == AggregateCountDistinct || ap.Opcode == AggregateSumDistinct
 }
 
+func (ap AggregateParams) changeField() bool {
```
Comment (systay) [inline go/vt/vtgate/engine/ordered_aggregate.go:72]: don't understand what check this is doing. could we add a comment to make it easier to understand?

File: go/vt/vtgate/engine/send.go [hunk 3]
```diff
@@ -151,10 +167,17 @@ func (s *Send) GetFields(vcursor VCursor, bindVars map[string]*querypb.BindVaria
...
 func (s *Send) description() PrimitiveDescription {
 	other := map[string]interface{}{
-		"Query":           s.Query,
-		"Table":           s.GetTableName(),
-		"IsDML":           s.IsDML,
-		"SingleShardOnly": s.SingleShardOnly,
+		"Query": s.Query,
+		"Table": s.GetTableName(),
+	}
+	if s.IsDML {
```
Comment (systay) [inline go/vt/vtgate/engine/send.go:173]: nice. much easier on the eyes

**Review by systay — APPROVED** [review]

**Comment by deepthi** [comment]
@harshit-gangal based on our discussion this morning, is the syntax going to change?

**Comment by harshit-gangal** [comment]
> @harshit-gangal based on our discussion this morning, isn't the syntax going to change?

We will be keeping this and also have one for shard GTIDs

**Comment by deepthi** [comment]
> > @harshit-gangal based on our discussion this morning, isn't the syntax going to change?
> 
> We will be keeping this and also have one for shard GTIDs

Then this doesn't really fix #7400, correct? Should that issue be re-opened?

**Pull request merged**